contracts/cw-core/src/contract.rs

```diff
@@ -687,9 +687,33 @@ pub fn query_cw20_balances(
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
-    // Don't do any state migrations.
-    Ok(Response::default())
+pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
+    match msg {
+        MigrateMsg::FromBeta {} => {
+            let keys_raw: Vec<(usize, Addr)> = PROPOSAL_MODULES
```

Comment:
this still does not deserialize properly into Addr

Reply:
aha it works now! needed to deserialize to string first then reconstruct address